src/bitcoin-cli.cpp

```diff
@@ -540,15 +540,15 @@ class NetinfoRequestHandler : public BaseRequestHandler
         // Report detailed peer connections list sorted by direction and minimum ping time.
         if (DetailsRequested() && !m_peers.empty()) {
             std::sort(m_peers.begin(), m_peers.end());
-            result += strprintf("<->   type   net  v  mping   ping send recv  txn  blk  hb %*s%*s%*s ",
-                                m_max_addr_processed_length, "addrp",
-                                m_max_addr_rate_limited_length, "addrl",
-                                m_max_age_length, "age");
+            result += tfm::format_raw("<->   type   net  v  mping   ping send recv  txn  blk  hb %*s%*s%*s ",
+                                      m_max_addr_processed_length, "addrp",
+                                      m_max_addr_rate_limited_length, "addrl",
+                                      m_max_age_length, "age");
```

Comment:
Is all the indentation churn really necessary?

Does changing strprintf to tfm_format_raw here catch any potential issues that wouldn't be seen in manual testing?

Reply:
> Is all the indentation churn really necessary?

It's `clang-format` output which I believe is what we're trying to adhere to.

> Does changing strprintf to tfm_format_raw here catch any potential issues that wouldn't be seen in manual testing?

The code does not compile with `strprintf` because this PR enforces compile-time checks, which fail because of the width specifiers in the format string (there's quite some discussion on these limitations in https://github.com/bitcoin/bitcoin/pull/30546). See also this excerpt from the PR description that addresses this:

> * rename the non-compile-time-validated `tfm::format` overloads to `tfm::format_raw`, so existing callsites by default use the safer `util::ConstevalFormatString` `tfm::format` overloads. Callsites that for some reason don't pass the compile-time checks (such as in `bitcoin-cli.cpp`) can use `tfm::format_raw`.

Reply:
Clang-format can be helpful for new code, but I don't think it's necessary to re-indent code for it that isn't otherwise touched -- that's just noisy churn.

Reply:
> Clang-format can be helpful for new code, but I don't think it's necessary to re-indent code for it that isn't otherwise touched -- that's just noisy churn.

I'm just going with our `clang-format` settings as defined in `src/.clang-format` - if you feel like those settings are suboptimal opening a separate pull/issue for that is probably more appropriate?

> Please provide an example of how this code could crash currently without this change?

The PR states that there is no behaviour change, so (unless there are run-time bugs we haven't seen yet) we can't make this code crash without this change. The point of this PR is to make the code more robust against future changes, where a new/modified format string can cause a run-time error, e.g. if not all code-paths are manually tested by reviewers (which is not unusual).

Reply:
> Please provide an example of how this code could crash currently without this change?

If you're asking about how such an error could sneak in without this PR, a change as simple adding a single extra `*` would lead to a run-time exception if no extra parameter is added:

```cpp
            result += tfm::format_raw("<->   type   net  v  mping   ping send recv  txn  blk  hb %**s%*s%*s ",
```

Reply:
Thanks for the example. As the primary author of this code, I think the run-time error would be more clear, though a compile-time error would catch it one CLI command earlier. I haven't found this kind of error to be an issue for me over the past half decade on this code.

Also, this code path is manually tested by myself, reviewers and users. Perhaps for that reason, and because it's non-critical code, a maintainer explicitly asked that no test coverage be written for it.

So either way is fine, I suppose, but this conflicts with https://github.com/bitcoin/bitcoin/pull/30930 so it's more an annoyance than helpful for me.

Reply:
> I'm just going with our clang-format settings as defined in src/.clang-format

It's not a hard requirement to retouch existing code with that tool, and I'd prefer avoiding the churn.

Reply:
> As the primary author of this code, I think the run-time error would be more clear

I'm sorry, what? For what reason would we not use the logic we already have to catch (some/most) of these format errors at compile-time, and defer them to run-time instead, relying on manual review to make sure each code-path is ran, potentially crashing user software? That seems like the opposite of security engineering best practices to me.

> a maintainer explicitly asked that no test coverage be written for it.

We're not writing any test coverage for it here either, it comes for free. How is that a bad thing?

> It's not a hard requirement to retouch existing code with that tool, and I'd prefer avoiding the churn.

Noted, but I prefer keeping this as is, I'd rather not override `clang-format` for this.

Reply:
> I'd prefer avoiding the churn.

We usually call things churn that change things multiple times within the same PR. Indenting the lines below a changed line that belong to the same logical unit is a good thing in my eyes. If we would not do that anywhere, the code would look weird and probably make `clang-format` unusable after a while.

Reply:
> The rest of this PR seems fine.

We can't have the rest of this PR without updating `bitcoin-cli.cpp`, unless someone implements the complete format string validation logic, for which as per my earlier comment, see https://github.com/bitcoin/bitcoin/pull/30546.

Reply:
> I'm sorry, what? For what reason would we not use the logic we already have to catch (some/most) of these format errors at compile-time, and defer them to run-time instead, relying on manual review to make sure each code-path is ran, potentially crashing user software? 

> relying on manual review to make sure each code-path is ran, potentially crashing user software

Because, as I wrote:

- I think the run-time error is more clear to understand, as the person working on the code
- This code path is manually tested anyway, otherwise the command with any non-zero level won't run

> That seems like the opposite of security engineering best practices to me.

One could argue the same for discouraging test coverage, but that's been the stance.

Reply:
> > I'd prefer avoiding the churn.
> 
> We usually call things churn that change things multiple times within the same PR. Indenting the lines below a changed line that belong to the same logical unit is a good thing in my eyes. If we would not do that anywhere, the code would look weird and probably make `clang-format` unusable after a while.

You're changing those lines needlessly. Whether you prefer to call it noise or churn is unimportant.

Reply:
@TheCharlatan mind being explicit what part of my feedback your emoji relates to?

Reply:
> I think the run-time error is more clear to understand, as the person working on the code

And I think prioritizing devs getting nicer error messages over increased software run-time stability is almost never a good trade-off. It seems we have fundamentally different views, so I'll leave this thread unresolved but I strongly disagree with your approach and won't change the PR because of it.

> This code path is manually tested anyway, otherwise the command with any non-zero level won't run

I understand it's tested, and that's good, because `tfm::format_raw()` doesn't do any compile-time checks.

Reply:
> mind being explicit what part of my feedback your emoji relates to?

Sure

> I haven't found this kind of error to be an issue for me over the past half decade on this code.

I've seen a bunch of positional args errors in fairly recent times, and catching them at compile time is certainly worthwhile. People tried doing this with various linters, but the linters produced false-positives, or missed certain cases.

Reply:
> And I think prioritizing devs getting nicer error messages over increased software run-time stability is almost never a good trade-off. It seems we have fundamentally different views

I think we agree there, given that you wrote "almost" ;)

Reply:
> Does changing strprintf to tfm_format_raw here catch any potential issues that wouldn't be seen in manual testing?

By manual testing do you mean automated testing of print statements?

> Clang-format can be helpful for new code, but I don't think it's necessary to re-indent code for it that isn't otherwise touched

Please format affected code, don't leave it unformatted, that's why the current code is so confusing.
```
Coding Style (General)
----------------------

Various coding styles have been used during the history of the codebase,
and the result is not very consistent. However, we're now trying to converge to
a single style, which is specified below. When writing patches, favor the new
style over attempting to mimic the surrounding style, except for move-only
commits.
```

> I haven't found this kind of error to be an issue for me over the past half decade on this code.

Others have found it to be an issue, better catch them as early as we can. Why wait until runtime or CI failure or release.

Reply:
There was no bikeshedding about this before on https://github.com/bitcoin/bitcoin/pull/31149, it just started.

Reply:
> By manual testing do you mean automated testing of print statements?

No, manual testing by building and running the CLI commands.

There is no automated test coverage of `-netinfo` (the touched code above), nor, for instance, of `-addrinfo`.

Reply:
By doing some of the validation at compile time, there **is** now some automated 'test coverage' on it - i.e. it will bite back if it's very wrong without any additional manual check.

Reply:
Generally unneeded changes can be avoided, but that was just a suggestion. Further historical context to get started:

- https://github.com/bitcoin/bitcoin/issues/18651
- https://github.com/bitcoin/bitcoin/issues/15465

Reply:
Useful context, thanks!
We must however keep our kitchen clean, we can't just be cooking all the time.
This change by @stickies-v was a small, low-risk change, making the code slightly more maintainable (or at least slightly less unmaintainable).
We should encourage these.